src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,560 @@
+CONFIG_UPS = {"name": "ups", "role": "ups"}
+
+CONFIG_ORSAY = {"name": "Orsay", "role": "orsay", "host": "192.168.56.101",
+                "children": {"pneumatic-suspension": CONFIG_PSUS,
+                             "pressure": CONFIG_PRESSURE,
+                             "pumping-system": CONFIG_PSYS,
+                             "ups": CONFIG_UPS}
+                }
+
+
+class TestOrsayStatic(unittest.TestCase):
+    """
+    Tests which don't need an Orsay component ready
+    """
+
+    def test_creation(self):
+        """
+        Test to create an Orsay component
+        """
+        try:
+            oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            self.fail(e)
+        self.assertEqual(len(oserver.children.value), 4)
+
+        oserver.terminate()
+
+
+class TestOrsay(unittest.TestCase):
+    """
+    Tests to run on the main Orsay component
+    """
+
+    def setUp(self):
+        """
+        Setup the Orsay client
+        """
+        try:
+            self.oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            self.skipTest(e)
+
+        for child in self.oserver.children.value:
+            if child.name == CONFIG_PSUS["name"]:
+                self.psus = child
+            elif child.name == CONFIG_PRESSURE["name"]:
+                self.pressure = child
+            elif child.name == CONFIG_PSYS["name"]:
+                self.psys = child
+            elif child.name == CONFIG_UPS["name"]:
+                self.ups = child
+
+    def tearDown(self):
+        """
+        Terminate the Orsay client
+        """
+        self.oserver.terminate()
+
+    def test_updateProcessInfo(self):
+        """
+        Check that the processInfo VA is updated properly and an exception is raised when the wrong parameter is passed
+        """
+        self.assertRaises(Exception, self.oserver._updateProcessInfo,
+                          {'parameter': self.oserver.datamodel.HybridPlatform.Cancel})
+        if not TEST_NOHW:
+            self.skipTest("TEST_NOHW is not set, cannot force data on Actual parameters of Orsay server outside of "
+                          "simulation")
+        test_string = "Some process information"
+        self.oserver.datamodel.HybridPlatform.ProcessInfo.Actual = test_string
+        self.assertEqual(self.oserver.processInfo.value, test_string)
+        self.oserver.datamodel.HybridPlatform.ProcessInfo.Actual = ""
+
+    def test_reconnection(self):
+        """
+        Checks that after reconnection things still work
+        """
+        self.oserver._device.HttpConnection.close()  # close the connection
+        self.oserver._device.MessageConnection.Connection.close()
+        self.oserver._device.DataConnection.Connection.close()
+        self.oserver._device.MessageConnection.dataConnection.Connection.close()
+        while not self.oserver.connected.is_set():
+            sleep(1)  # wait for the reconnection
```

Comment:
Don't rely on non official API in test cases.
I think it'd be fine to just wait "long enough" and try again.

Possibly, you could first check that the .state indicates a HwError, and that afterwards, it goes back to ST_RUNNING. (However, that's not super easy as probably the reconnection will be very fast, with the test case, so you'd need to subscribe to the state, and record all the values passed to the subscriber).